routers/repo/issue.go

```diff
@@ -1467,3 +1467,65 @@ func ChangeCommentReaction(ctx *context.Context, form auth.ReactionForm) {
 		"html": html,
 	})
 }
+
+// UpdateDeadline adds or updates a deadline
+func UpdateDeadline(ctx *context.Context, form auth.DeadlineForm) {
+	issue := GetActionIssue(ctx)
+	if ctx.Written() {
+		return
+	}
+
+	if ctx.HasError() {
+		ctx.ServerError("ChangeIssueDeadline", errors.New(ctx.GetErrMsg()))
+		return
+	}
+
+	// Check if the user has at least write access to the repo
+	if !ctx.Repo.IsWriter() {
```

Comment:
I found there is authorization middleware, can you use that instead? Like in `m.Post("/status", reqRepoWriter, repo.UpdateIssueStatus)`. It calls `RequireRepoWriter()` func.

Reply:
Done.